File: pool_alerts.ts
#!/usr/bin/env node
/**
 * Pool Alert System - TypeScript
 * Sends alerts when pools meet certain criteria
 */

import axios from 'axios';

interface Pool {
  address?: string;
  name?: string;
  apr?: number;
  farmApr?: number;
  volume24h?: number;
  volume?: number;
  tvl?: number;
}

interface AlertConfig {
  minApr: number;
  minVolume: number;
}

const METEORA_API = 'https://api.meteora.ag';

async function fetchPools(): Promise<Pool[]> {
  try {
    const response = await axios.get(`${METEORA_API}/pools`, { timeout: 30000 });
    return response.data || [];
  } catch (error) {
    console.error('Error fetching pools:', error);
    return [];
  }
}

function checkAlerts(pools: Pool[], config: AlertConfig): Pool[] {
  return pools.filter((pool) => {
    const apr = pool.apr || pool.farmApr || 0;
    const volume = pool.volume24h || pool.volume || 0;
    return apr >= config.minApr && volume >= config.minVolume;
  });
}

async function main() {
  const args = process.argv.slice(2);
  
  const getArg = (flag: string, defaultValue: number): number => {
    const index = args.indexOf(flag);
    return index !== -1 ? parseFloat(args[index + 1]) : defaultValue;
  };

  const minApr = getArg('--min-apr', 20);
  const minVolume = getArg('--min-volume', 100000);

  console.log(`🔍 Checking for pools with APR >= ${minApr}% and Volume >= $${minVolume.toLocaleString()}`);
  console.log(`Time: ${new Date().toISOString()}\n`);

  const pools = await fetchPools();
  if (!pools.length) {
    console.log('❌ Could not fetch pools');
    return;
  }

  const alerts = checkAlerts(pools, { minApr, minVolume });

  if (alerts.length) {
    console.log(`🚨 FOUND ${alerts.length} ALERTS:\n`);
    alerts.forEach((pool) => {
      const apr = pool.apr || pool.farmApr || 0;
      const volume = pool.volume24h || pool.volume || 0;
      const tvl = pool.tvl || 0;
      
      console.log(`⚡ ${pool.name || 'Unknown Pool'}`);
      console.log(`   APR: ${apr.toFixed(2)}%`);
      console.log(`   Volume 24h: $${volume.toLocaleString()}`);
      console.log(`   TVL: $${tvl.toLocaleString()}`);
      console.log(`   Address: ${pool.address || pool.name || 'N/A'}`);
      console.log();
    });
  } else {
    console.log('✅ No pools meet alert criteria');
  }
}

main().catch(console.error);

File: pool_alerts.py
#!/usr/bin/env python3
"""
Meteora Pool Alert System
Sends alerts when pools meet certain criteria.
Usage: python3 pool_alerts.py --min-apr 20 --min-volume 100000
"""

import argparse
import requests
import json
import sys
from datetime import datetime

METEORA_API = "https://api.meteora.ag"

def fetch_pools():
    """Fetch pool data from Meteora."""
    try:
        response = requests.get(f"{METEORA_API}/pools", timeout=30)
        response.raise_for_status()
        return response.json()
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)
        return []

def check_alerts(pools, min_apr, min_volume):
    """Check for pools meeting alert criteria."""
    alerts = []
    
    for pool in pools:
        apr = pool.get('apr', 0) or pool.get('farmApr', 0) or 0
        volume = pool.get('volume24h', 0) or pool.get('volume', 0) or 0
        tvl = pool.get('tvl', 0) or 0
        
        if apr >= min_apr and volume >= min_volume:
            alerts.append({
                'pool': pool,
                'apr': apr,
                'volume': volume,
                'tvl': tvl
            })
    
    return alerts

def main():
    parser = argparse.ArgumentParser(description="Meteora Pool Alert System")
    parser.add_argument("--min-apr", type=float, default=20, help="Minimum APR %")
    parser.add_argument("--min-volume", type=float, default=100000, help="Minimum 24h volume")
    
    args = parser.parse_args()
    
    print(f"🔍 Checking for pools with APR >= {args.min_apr}% and Volume >= ${args.min_volume:,.0f}")
    print(f"Time: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\n")
    
    pools = fetch_pools()
    if not pools:
        print("❌ Could not fetch pools")
        return
    
    alerts = check_alerts(pools, args.min_apr, args.min_volume)
    
    if alerts:
        print(f"🚨 FOUND {len(alerts)} ALERTS:\n")
        for alert in alerts:
            pool = alert['pool']
            print(f"⚡ {pool.get('name', 'Unknown Pool')}")
            print(f"   APR: {alert['apr']:.2f}%")
            print(f"   Volume 24h: ${alert['volume']:,.2f}")
            print(f"   TVL: ${alert['tvl']:,.2f}")
            print(f"   Address: {pool.get('address', pool.get('pool', 'N/A'))}")
            print()
    else:
        print("✅ No pools meet alert criteria")

if __name__ == "__main__":
    main()
